lib/Doctrine/Migrations/Configuration/Loader/YamlFileLoader.php

```diff
@@ -2,33 +2,46 @@
 
 declare(strict_types=1);
 
-namespace Doctrine\Migrations\Configuration;
+namespace Doctrine\Migrations\Configuration\Loader;
 
+use Doctrine\Migrations\Configuration\Configuration;
+use Doctrine\Migrations\Configuration\Exception\FileNotFound;
 use Doctrine\Migrations\Configuration\Exception\YamlNotAvailable;
 use Doctrine\Migrations\Configuration\Exception\YamlNotValid;
 use Symfony\Component\Yaml\Exception\ParseException;
 use Symfony\Component\Yaml\Yaml;
 use function assert;
 use function class_exists;
+use function file_exists;
 use function file_get_contents;
 use function is_array;
 
 /**
- * The YamlConfiguration class is responsible for loading migration configuration information from a YAML file.
- *
  * @internal
  */
-class YamlConfiguration extends AbstractFileConfiguration
+final class YamlFileLoader extends AbstractFileLoader
 {
+    /** @var ArrayLoader */
+    private $arrayLoader;
+
+    public function __construct(?ArrayLoader $arrayLoader = null)
```

Comment:
It's obviously a hard dependency, what do you think about making that clear by removing both the

- nullable type declaration and the
- default value

?